src/System.Memory/tests/Memory/TryGetArray.cs

```diff
@@ -45,5 +45,39 @@ public static void OwnedMemoryTryGetArray()
                 Assert.Equal(array[i], segment.Array[i]);
             }
         }
+        
+        [Fact]
+        public static void TryGetArrayFromEmptyMemory()
+        {
+            int[] array = new int[0];
+            Memory<int> memory = array;
+
+            Assert.True(memory.TryGetArray(out ArraySegment<int> segment));
+            Assert.Same(array, segment.Array);
+            Assert.Equal(0, segment.Array.Length);
+
+            Assert.True(Memory<byte>.Empty.TryGetArray(out ArraySegment<byte> byteSegment));
+        }
+
+        [Fact]
+        public static void TryGetArrayFromEmptyOwnedMemory()
+        {
+            int[] array = new int[0];
+            OwnedMemory<int> owner = new CustomMemoryForTest<int>(array);
+
+            Assert.True(owner.Memory.TryGetArray(out ArraySegment<int> segment));
+            Assert.Same(array, segment.Array);
+            Assert.Equal(0, segment.Array.Length);
+        }
+
+        [Fact]
+        public static void TryGetArrayFromEmptyMemoryAndNegativeIndex()
+        {
+            using (var memory = new NativeOwnedMemory(42))
+            {
+                Assert.True(memory.TryGetArray(out ArraySegment<byte> segment));
```

Comment:
This will be Assert.False.

It will only be true if NativeOwnedMemory.Length == 0.

Reply:
Changed length from 42 to 0, we want to test the empty case.